Hospital/HospitalApi/Controllers/MedicationExpenditureReportController.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Hospital.MedicalRecords.Model;
+using Hospital.MedicalRecords.Service;
+using Hospital.SharedModel.Model.Wrappers;
+using Hospital.SharedModel.Repository.Base;
+using HospitalApi.DTOs;
+
+namespace HospitalApi.Controllers
+{
+    [Route("api/[controller]/[action]")]
+    [ApiController]
+    public class MedicationExpenditureReportController : ControllerBase
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        public MedicationExpenditureReportController(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+        }
+
+        [HttpPost]
+        public IActionResult GetMedicationExpenditureReport(TimePeriod timePeriod)
+        {
+            if (timePeriod.EndTime == null || timePeriod.StartTime == null || timePeriod.StartTime > timePeriod.EndTime)
```

Comment:
Instead of checking wheter DateTime obj is null try this 'datetime == DateTime.MinValue' or 'datetime.HasValue'.